src/status_im/contexts/wallet/wallet_connect/events/session_proposals.cljs

```diff
@@ -135,34 +135,60 @@
                  :proposal-request current-proposal
                  :session-networks session-networks
                  :address          current-address
-                 :on-success       #(rf/dispatch [:wallet-connect/approve-session-success %])
-                 :on-fail          #(rf/dispatch [:wallet-connect/approve-session-error %])}])
+                 :on-success       [:wallet-connect/approve-session-success current-proposal]
+                 :on-fail          [:wallet-connect/approve-session-error current-proposal]}])
              [:dispatch [:dismiss-modal :screen/wallet.wallet-connect-session-proposal]]]}
        {:fx [[:dispatch [:wallet-connect/no-internet-toast]]]}))))
 
 (rf/reg-event-fx :wallet-connect/approve-session-success
- (fn [_ [session]]
-   (log/info "Wallet Connect session approved")
-   {:fx [[:dispatch [:wallet-connect/on-new-session session]]
-         [:dispatch [:wallet-connect/reset-current-session-proposal]]
-         [:dispatch [:wallet-connect/redirect-to-dapp (data-store/get-dapp-redirect-url session)]]]}))
+ (fn [{:keys [db]} [proposal session]]
+   (let [networks              (data-store/get-networks-from-proposal db proposal)
+         total-connected-dapps (-> db
+                                   :wallet-connect/sessions
+                                   count
+                                   inc)]
+     {:fx [[:dispatch [:wallet-connect/on-new-session session]]
+           [:dispatch [:wallet-connect/reset-current-session-proposal]]
+           [:dispatch [:wallet-connect/redirect-to-dapp (data-store/get-dapp-redirect-url session)]]
+           [:dispatch
+            [:centralized-metrics/track :metric/dapp-session
+             {:networks networks
+              :action   :approved
+              :result   :success}]]
+           [:dispatch
+            [:centralized-metrics/track :metric/dapp-connected
+             {:networks              networks
+              :total_connected_dapps total-connected-dapps}]]]})))
```

Comment:
from BA perspective if you are looking for just connected dapp, dapp-connected would be enough. but dapp-session it's about all proposal request that will be approved/rejected success/fail.

Reply:
but if we look at the approved `dapp-session` events, we will get the same thing as `dapp-connected`, no? They, do the same thing it seems, just `dapp-connected` sends an extra `:total_connected_dapps` key.

Reply:
yeah but you should filter dapp-session to action:approved, result:sucess to get list of all dapp connected. do we have any limitation regarding event number (I understand we try to avoid adding not useful event)? 
also as we have dapp-disconnected, i though it will be good to have another event as dapp-connected to give you information straight away.

Reply:
There is no limit on the number of events, but we should be careful and not collect duplicates. 

If some information can be inferred from given data, we should avoid collecting it. 

Is approval is enough for us to know that the dapp was connected ?
And maybe we send `total-connected-dapps` in the approval event itself ?

Reply:
If we are speaking of unique event names, there's a soft limit of 5000 
https://docs.mixpanel.com/docs/data-structure/events-and-properties#what-are-the-limits-of-events-and-properties. We are far from reaching this number, but we should be mindful.